pygpudrive/env/config.py

```diff
@@ -1,83 +1,66 @@
-    # # VALUES BELOW ARE ENV CONSTANTS: DO NOT CHANGE # #
-    # # # # # # # # # # # # # # # # # # # # # # # # # # #
-    # Normalization constants
-    max_speed: int = 100
-    max_veh_len: int = 30
-    max_veh_width: int = 10
-    min_rel_goal_coord: int = -1000
-    max_rel_goal_coord: int = 1000
-    min_rel_agent_pos: int = -1000
-    max_rel_agent_pos: int = 1000
-    max_orientation_rad: float = 2 * np.pi
-    min_rm_coord: int = -1000
-    max_rm_coord: int = 1000
-    max_road_line_segmment_len: int = 100
-    max_road_scale: int = 100
-
-    # Feature dimensions
-    EGO_STATE_DIM = 6 if ego_state else 0
-    PARTNER_DIM = 10 if partner_obs else 0
-    ROAD_MAP_DIM = 13 if road_map_obs else 0
-    
-    # Agent tensor shape set in consts.hpp (kMaxAgentCount)
-    k_max_agent_count: int = 128
-    
+    """Configuration settings for the GPUDrive gym environment.
+
+    This class contains both Python-specific configurations and settings that
+    are shared between Python and C++ components of the simulator. 
+
+    To modify simulator settings shared with C++, follow these steps:
+    1. Navigate to `src/consts.hpp` in the C++ codebase.
+    2. Locate and modify the desired constant definitions (e.g., `maxNumObjects`).
+    3. Save the changes to `src/consts.hpp`.
+    4. Recompile the simulator to apply changes across both C++ and Python environments.
+    """
+
+    # Python-specific configurations
+    # Observation space settings
+    ego_state: bool = True  # Include ego vehicle state in observations
+    road_map_obs: bool = True  # Include road graph in observations
+    partner_obs: bool = True  # Include partner vehicle info in observations
+    norm_obs: bool = True  # Normalize observations
+    enable_lidar: bool = False  # Enable LiDAR data in observations
+
+    # Road observation algorithm settings
+    road_obs_algorithm: str = "linear"  # Algorithm for road observations
+    obs_radius: float = 100.0  # Radius for road observations
+    polyline_reduction_threshold: float = 1.0  # Threshold for polyline reduction
+
+    # Action space settings (joint discrete)
+    steer_actions: torch.Tensor = torch.round(torch.linspace(-1.0, 1.0, 13), decimals=3)
+    accel_actions: torch.Tensor = torch.round(torch.linspace(-4.0, 4.0, 7), decimals=3)
+
+    # Collision behavior settings
+    collision_behavior: str = "remove"  # Options: "remove", "stop", "ignore"
+
+    # Scene configuration
+    remove_non_vehicles: bool = True  # Remove non-vehicle entities from the scene
+
+    # Reward settings
+    reward_type: str = "sparse_on_goal_achieved"  # Options: "sparse_on_goal_achieved"
+    dist_to_goal_threshold: float = 3.0  # Radius around goal considered as "goal achieved"
+
+    # Feature dimensions (automatically calculated based on configurations)
+    EGO_STATE_DIM: int = 6 if ego_state else 0
+    PARTNER_DIM: int = 10 if partner_obs else 0
+    ROAD_MAP_DIM: int = 13 if road_map_obs else 0
```

Comment:
I am unsure on how these dims are automatically calculated. I dont see where they are set in code.